### DISCUSSION TITLE
[BUG] Bare Metal Installation on Debian Buster

### DISCUSSION THREAD

**Discussion by steviehs** [jonaswinkler/paperless-ng#1021] [Support] [answered]
I tried to do a fresh bare metal install on Buster. Unfortunately pikepdf seems to require a newer libqpdf than installed on Buster. Any hints how to get along with that?

    src/qpdf/qpdf.cpp:892:65: error: ‘ownerPasswordMatched’ is not a member of ‘QPDF’
             .def_property_readonly("owner_password_matched", &QPDF::ownerPasswordMatched,
                                                                     ^~~~~~~~~~~~~~~~~~~~

**Comment by jonaswinkler** [comment]
What architecture are you on? If you use precompiled packages of pikepdf, the libqpdf dev files should not be required.

**Comment by steviehs** [comment]
I am on X86, basically I followed the instructions for bare metal which recommends to install pikepdf via pip, not via package management. 
What I could try now is to select every package, which is available as native .deb package and install that and try again with the rest from requirements.txt... as this is some work: is this the way to go?

**Reply by jonaswinkler** [reply]
No, that's not the way to go.

If I do `python3 -m pip install --upgrade pikepdf` on my amd64 debian buster machine, it installs without any issues, and I don't have any qpdf packages installed.

pip tries to install precompiled packages (these are called wheels in the python world) from pypi first. However, a wheel has to be compiled for each python version and architecture separately. If none is available, pip pulls the source code of the package and starts building. This is what happens here.

**Reply by steviehs** [reply]
root@home:/opt/paperless-ng# python3 --version
Python 3.7.3
I uninstalled qpdf and libqpdf and pip still tries to build pikepdf... strange...

**Comment by steviehs** [comment, marked as answer]
ok, got it: 
sudo -Hu paperless pip3 install --upgrade pip
updates pip first on buster and then all requirements are installed from wheel. 
I have added this to the documentation. Thanks for your patience :-)
